cmd/zoekt-sourcegraph-indexserver/cleanup.go

```diff
@@ -172,6 +158,38 @@ func cleanup(indexDir string, repos []uint32, now time.Time, shardMerging bool)
 		}
 	}
 
+	// remove any Zoekt metadata files in the given dir that don't have an
+	// associated shard file
```

Comment:
No, I added this functionality since we switched the implementation. Before, the implementation made sure to delete the metadata file _before_ its associated shard file. Since we switched to using zoekt.IndexFilePaths that order isn't guaranteed anymore (leaving open the possibility of a metadata file not having an associated shard if we delete the shard and then crash). I added this additional logic to cleanup so that we would have a background process that would reap those "stranded" files. 

I am happy to pull bit into a separate PR though.